website_sale_product_search_advanced/models/product_template.py

```diff
@@ -0,0 +1,19 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    website_partner_ref = fields.Char(
+        string='Partner Reference'
+    )
+
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        res = super()._search_get_detail(website, order, options)
+        res['search_fields'].append('barcode')
+        res['search_fields'].append('website_partner_ref')
```

Comment:
In original _search_get_detail code you can see that not only desired fields are added to `search_fields` key, but also to `mapping` clause (not sure which is the need), and even in `fetch_fields` (in some cases, indeed). Could you investigate a little but about those keys and their usage?

Reply:
- search_fields: fields within which the search term must be found
            - fetch_fields: fields from which data must be fetched